scripts/bootstrap.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python2
+"""
+Reconfigures a Jenkins master running in Docker at container runtime.
+"""
+
+from __future__ import print_function
+import os
+import re
+import subprocess
+import sys
+import xml.etree.ElementTree as ET
+
+
+def is_firstrun(jenkins_home_dir):
+    """A small helper utility to determine if this is the first run of this
+    bootstrap script. Checks to see if the 'jenkins_home_dir' directory
+    is empty, or if it contains existing data.
+
+    :param jenkins_home_dir: the path to $JENKINS_HOME on disk
+    :return: boolean; True if $JENKINS_HOME isn't populated, false otherwise
+    """
+    if len(os.listdir(jenkins_home_dir)) == 0:
+        return True
+    else:
+        return False
+
+
+def populate_jenkins_config_xml(config_xml, name, host, port, context):
+    """Modifies a Jenkins master's 'config.xml' at runtime. Essentially, this
+    replaces certain configuration options of the Mesos plugin, such as the
+    framework name and the Jenkins URL that agents use to connect back to the
+    master.
+
+    :param config_xml: the path to Jenkins' 'config.xml' file
+    :param name: the name of the framework, e.g. 'jenkins'
+    :param host: the Mesos agent the task is running on
+    :param port: the Mesos port the task is running on
+    :param context: the context of the Jenkins application, e.g.
+        '/service/jenkins'
+    """
+    tree = ET.parse(config_xml)
+    root = tree.getroot()
+    mesos = root.find('./clouds/org.jenkinsci.plugins.mesos.MesosCloud')
+
+    mesos_frameworkName = mesos.find('./frameworkName')
+    mesos_frameworkName.text = name
+
+    mesos_jenkinsURL = mesos.find('./jenkinsURL')
+    mesos_jenkinsURL.text = ''.join(['http://', host, ':', port, context])
```

Comment:
Good catch, it looks like when I rebased I lost a commit. Let me fix that up quick.